restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/AccountsEndpoint.java

```diff
@@ -597,6 +583,118 @@ protected Response updateAccount(final String identifier, final MultivaluedMap<S
         }
     }
 
+    private Account getOperatingAccount (String identifier) {
+        Sid sid = null;
+        Account account = null;
+        try {
+            sid = new Sid(identifier);
+            account = accountsDao.getAccount(sid);
+        } catch (Exception e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Exception trying to get account using SID. Seems we have email as identifier");
+            }
+        }
+        if (account == null) {
+            account = accountsDao.getAccount(identifier);
+        }
+        return account;
+    }
+
+    private Organization getOrganization(final MultivaluedMap<String, String> data) {
+        Organization organization = null;
+        Sid organizationSid = null;
+        String organizationId = null;
+
+        if (data.containsKey("Organization")) {
+            organizationId = data.getFirst("Organization");
+        } else {
+            return null;
+        }
+
+        //Attempt to get Organization by domain name
+        organization = organizationsDao.getOrganizationByDomainName(organizationId);
+
+        if (organization != null) {
+            return organization;
+        } else {
+            //Attempt to match organizationId to organizationSid
+            try {
+                organizationSid = new Sid(organizationId);
```

Comment:
Suggestion:can we do this first to check whether this is domain or Sid. If you can create a Sid form it, we know for sure its an Sid, if not, we assume is a domain (which validation is more opened). This may optimize the number of queries required....

Reply:
@jaimecasero sure I can modify this to improve the method